### PR TITLE
Relay 100 split aggregator

### PR DESCRIPTION
**Pull request by ilyasymbiotic**
### **PR Type**
Enhancement, Tests

___

### **Description**
• **Major refactoring**: Introduced `ValidatorMap` entity for centralized quorum tracking and validator presence management
• **Aggregator improvements**: Refactored aggregator app to use `ValidatorMap` for simplified quorum checking instead of manual validator lookup
• **New signature listener**: Added dedicated signature listener use case for handling incoming P2P signature messages with transaction-based processing
• **Repository enhancements**: Implemented BadgerDB storage for `ValidatorMap` with JSON serialization and added transaction helper methods (`DoUpdateInTx`, `DoViewInTx`) across all repository modules
• **Comprehensive testing**: Added extensive unit tests for all new components including `ValidatorMap`, aggregator app, signature listener, and repository operations with real cryptographic integration
• **Type system improvements**: Updated signature handling to use typed `RawPublicKey` instead of raw bytes and moved aggregation status to core entity package
• **Configuration cleanup**: Moved config types from internal entity to local definitions and integrated signature listener into main application flow

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["P2P Signature Messages"] --> B["Signature Listener"]
  B --> C["ValidatorMap"]
  C --> D["Aggregator App"]
  D --> E["Quorum Check"]
  F["BadgerDB Repository"] --> C
  C --> F
  G["Transaction Helpers"] --> F
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><details><summary>9 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>entity_validator_map_test.go</strong><dd><code>Add comprehensive ValidatorMap unit tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

core/entity/entity_validator_map_test.go

• Added comprehensive unit tests for <code>ValidatorMap</code> functionality <br>including creation, validator presence tracking, and threshold <br>checking<br> • Tests cover edge cases like empty validator sets, inactive <br>validators, and large number scenarios<br> • Includes integration tests <br>for realistic quorum scenarios with different threshold conditions

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-785b143de2b404500adcb5616df8b9759262f2cd3639b23aea73b9929782a28d">+483/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>badger_repository_validator_map_test.go</strong><dd><code>Add BadgerDB ValidatorMap repository tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/client/repository/badger/badger_repository_validator_map_test.go

• Added unit tests for BadgerDB repository operations for <code>ValidatorMap</code> <br>storage and retrieval<br> • Tests serialization/deserialization with <br>various data scenarios including empty maps and large numbers<br> • <br>Includes transaction testing for rollback scenarios and nested <br>transaction behavior

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-6f45c7f57e2ef4d3699babc49a40518cd757efee5ad4d781673be68b77ac3f21">+470/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>signer_app.go</strong><dd><code>Add generated mocks for signer app interfaces</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/usecase/signer-app/mocks/signer_app.go

• Generated mock implementations for signer app interfaces using <br>MockGen<br> • Includes mocks for repository, p2p service, key provider, <br>aggregation proof signal, aggregator, and metrics interfaces

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-003d3d91b36347f5f0eeadb7f09c29dcf4f12297c926dcea9ed3e3cdbf24ee56">+376/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>badger_repository_transaction_test.go</strong><dd><code>Add BadgerDB transaction management tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/client/repository/badger/badger_repository_transaction_test.go

• Added comprehensive tests for BadgerDB transaction management <br>including <code>DoUpdateInTx</code> functionality<br> • Tests transaction rollback <br>behavior, nested transactions, and read-only transaction constraints<br> • <br>Verifies atomic operations and proper transaction context handling

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-69afcaaff629760fd672289fcb15cba32ebaec975416c0a593b0abe452cf06d0">+211/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>aggregator_app.go</strong><dd><code>Update aggregator app mocks for ValidatorMap integration</code>&nbsp; </dd></summary>
<hr>

internal/usecase/aggregator-app/mocks/aggregator_app.go

• Updated mock repository interface by replacing <code>GetValidatorByKey</code> and <br><code>SaveSignature</code> with <code>GetValidatorMap</code><br> • Added new <code>Mockaggregator</code> <br>interface with <code>Aggregate</code>, <code>GenerateExtraData</code>, and <code>Verify</code> methods

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-85ed08da6c74966c5bfa6bf4f82199d73ee22ee7fa5de0a3d66e87a5841483e7">+76/-21</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>aggregator_app_test.go</strong><dd><code>Add aggregator app unit tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/usecase/aggregator-app/aggregator_app_test.go

• Added unit tests for aggregator app covering quorum reached and not <br>reached scenarios<br> • Tests signature message handling with proper mock <br>setup for dependencies<br> • Verifies correct behavior when threshold <br>conditions are met or not met

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-374236a48b4628a2d61f215bb4ec6e5450f56285ec3b40ddfd280514b5e442cc">+165/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>signature_listener_uc.go</strong><dd><code>Add generated mocks for signature listener interfaces</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/usecase/signature-listener/mocks/signature_listener_uc.go

• Generated mock implementations for signature listener use case <br>interfaces<br> • Includes mocks for repository operations like <br><code>GetValidatorByKey</code>, <code>UpdateValidatorMap</code>, and transaction handling

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-d91700762a254499e5c699d751e9294125fd0b7d67c4b652a5e99b16d596cf82">+130/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>signer_app_test.go</strong><dd><code>Add signer app unit tests with crypto integration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/usecase/signer-app/signer_app_test.go

• Added unit tests for signer app functionality with real <br>cryptographic operations<br> • Tests signature generation, validation, and <br>storage using actual private keys<br> • Verifies integration with <br>repository and validator set management

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-55eb4622bf16d7eb4d6b23fbea6b071a3f232d2a023cb0f73f4bf98e32bf0692">+151/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>signature_listener_uc_test.go</strong><dd><code>Add signature listener use case tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/usecase/signature-listener/signature_listener_uc_test.go

• Added unit tests for signature listener use case with real <br>cryptographic signatures<br> • Tests complete flow from P2P message <br>reception to validator map updates<br> • Verifies signature validation and <br>storage with actual private key operations

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-7a883c583e5937c16e0e8e840ff4c11e480833fd3d1899b8f36178ec4351fd88">+143/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>19 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>aggregator_app.go</strong><dd><code>Refactor aggregator to use ValidatorMap for quorum tracking</code></dd></summary>
<hr>

internal/usecase/aggregator-app/aggregator_app.go

• Refactored signature handling to use <code>ValidatorMap</code> for quorum <br>tracking instead of manual validator lookup<br> • Removed direct validator <br>verification and signature saving logic, delegating to validator map<br> • <br>Simplified quorum checking by using <code>ValidatorMap.ThresholdReached()</code> <br>method<br> • Updated interface to use local <code>aggregator</code> interface instead <br>of external package

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-46cf96da1f87c1ed42ba9aeb106f594d2ccfd9f7b782622337709ca9bcf61d37">+36/-83</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>badger_repository_validator_set.go</strong><dd><code>Update validator set repository to use transaction contexts</code></dd></summary>
<hr>

internal/client/repository/badger/badger_repository_validator_set.go

• Updated all validator set repository methods to use transaction <br>context helpers (<code>DoUpdateInTx</code>, <code>DoViewInTx</code>)<br> • Replaced direct BadgerDB <br>transaction calls with context-aware transaction management<br> • <br>Maintains same functionality while improving transaction handling <br>consistency

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-faf4b448b8fc65cc49ee94afc9782e1d86f529e24558ea392f39ac867b046243">+18/-12</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>signature_listener_uc.go</strong><dd><code>Add signature listener use case for message processing</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/usecase/signature-listener/signature_listener_uc.go

• Created new signature listener use case for handling incoming <br>signature messages<br> • Implements validator verification, signature <br>validation, and <code>ValidatorMap</code> updates<br> • Manages transaction-based <br>operations for atomic signature processing and validator presence <br>tracking

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-49a39689b9881e49aeed3e5efe09cce911ae9dd59d215f08a7a1850470f3c765">+117/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>badger_repository_validator_map.go</strong><dd><code>Add BadgerDB ValidatorMap repository implementation</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/client/repository/badger/badger_repository_validator_map.go

• Implemented BadgerDB storage operations for <code>ValidatorMap</code> entities<br> • <br>Added serialization/deserialization using JSON with proper address and <br>big integer handling<br> • Includes transaction-aware CRUD operations with <br>proper error handling

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-07282dc64e8b45fcd20e6c8f9e3b9b389a7b97fa787fc5e8184b98c82045231e">+135/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>badger_repository_signature_request.go</strong><dd><code>Update signature request repository transaction handling</code>&nbsp; </dd></summary>
<hr>

internal/client/repository/badger/badger_repository_signature_request.go

• Updated signature request repository methods to use transaction <br>context helpers<br> • Replaced direct BadgerDB transaction calls with <br><code>DoUpdateInTx</code> and <code>DoViewInTx</code> patterns<br> • Maintains existing <br>functionality while improving transaction consistency

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-2021ea2acd66040976cb26ff27f35e109e0037a4e3df435c8a7487f84adf71cb">+9/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>badger_repository_network_config.go</strong><dd><code>Update network config repository transaction handling</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/client/repository/badger/badger_repository_network_config.go

• Updated network config repository methods to use transaction context <br>helpers<br> • Replaced direct BadgerDB operations with context-aware <br>transaction management<br> • Consistent with other repository modules for <br>transaction handling

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-c0eb33da31de04c647b83041d07ef4bc7c197bb5eb273d2c93aa26dfb7a2f8b0">+6/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>badger_repository_signature.go</strong><dd><code>Refactor signature repository to use typed keys and transaction </code><br><code>helpers</code></dd></summary>
<hr>

internal/client/repository/badger/badger_repository_signature.go

• Changed parameter type from <code>[]byte</code> to <code>entity.RawPublicKey</code> in <br>signature-related functions<br> • Replaced direct database transaction <br>calls with <code>DoUpdateInTx</code> and <code>DoViewInTx</code> methods<br> • Added context usage <br>by retrieving transaction from context using <code>getTxn(ctx)</code>

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-d8f9032067c85d71668da77e62f7cf51c9625e785efcfd5f68691c8984e7fd58">+7/-5</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>root.go</strong><dd><code>Integrate signature listener for message handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

cmd/relay/root.go

• Added import for <code>signatureListener</code> package<br> • Created new signature <br>listener instance and started signature message listener<br> • Moved <br>signature message handling from p2p service to signature listener with <br>saved message listener

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-c1698ba36d063bfa002e73dd2d0a518c17e10cbcb5b4895f3a9b31864bc9dbdb">+12/-2</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>badger_repository_pending_validator_set.go</strong><dd><code>Refactor pending validator set repository to use transaction helpers</code></dd></summary>
<hr>

internal/client/repository/badger/badger_repository_pending_validator_set.go

• Replaced direct database transaction calls with <code>DoUpdateInTx</code> and <br><code>DoViewInTx</code> methods<br> • Added context usage by retrieving transaction <br>from context using <code>getTxn(ctx)</code>

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-e6ba8bd7eab0a6f5c54572effc9f18ad7ada6a156b9513bd81e2ae305bed6fce">+6/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>badger_repository_aggregation_proof.go</strong><dd><code>Refactor aggregation proof repository to use transaction helpers</code></dd></summary>
<hr>

internal/client/repository/badger/badger_repository_aggregation_proof.go

• Replaced direct database transaction calls with <code>DoUpdateInTx</code> and <br><code>DoViewInTx</code> methods<br> • Added context usage by retrieving transaction <br>from context using <code>getTxn(ctx)</code>

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-da59e69cb72ea65614d7d9573ac77efcfd10c2b296f96bceba75d7187cb6509b">+6/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>entity_validator_map.go</strong><dd><code>Add ValidatorMap entity for tracking validator presence and voting </code><br><code>power</code></dd></summary>
<hr>

core/entity/entity_validator_map.go

• Added new <code>ValidatorMap</code> struct with request hash, epoch, validator <br>mappings and voting power tracking<br> • Implemented <code>NewValidatorMap</code> <br>constructor and methods for validator presence and threshold checking<br> <br>• Added functionality to track active validators and current voting <br>power against quorum threshold

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-e21e36841212084456bd2c474f48c06da4b61c6395d0b0ab62b91115a688be45">+54/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>signer_app.go</strong><dd><code>Add mock generation and update signature key type</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/usecase/signer-app/signer_app.go

• Added mockgen directive for generating mocks<br> • Changed <code>SaveSignature</code> <br>method parameter type from <code>[]byte</code> to <code>entity.RawPublicKey</code>

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-0544b8fb7016065ff2de8522f81490696d1e5b79cd86efa32d652a7b865bff41">+3/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>app.go</strong><dd><code>Refactor aggregation status to use core entity type</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/usecase/api-server/app.go

• Removed import of <code>p2pEntity</code> package<br> • Changed <code>GetAggregationStatus</code> <br>return type from <code>p2pEntity.AggregationStatus</code> to <br><code>entity.AggregationStatus</code>

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-b2c8c5eee668400ec6d5e50059cd926595235be3b685ecb54e512e7ba7c60473">+1/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>badger_repository_transaction.go</strong><dd><code>Add transaction helper methods for context-based transaction </code><br><code>management</code></dd></summary>
<hr>

internal/client/repository/badger/badger_repository_transaction.go

• Added new transaction helper methods <code>DoUpdateInTx</code> and <code>DoViewInTx</code><br> • <br>Implemented context-based transaction management with <code>getTxn</code> helper <br>function<br> • Added support for nested transactions by checking existing <br>transaction in context

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-db429a6f11f434467e51f7b6d5599bbe191ade238da52b8274fc80a5157ec912">+40/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>app_mock.go</strong><dd><code>Update mock to use core entity aggregation status type</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/usecase/api-server/mocks/app_mock.go

• Removed import of <code>entity0</code> (p2pEntity) package<br> • Updated mock method <br>to return <code>entity.AggregationStatus</code> instead of <br><code>entity0.AggregationStatus</code>

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-86352c5516df269d1c9a16d9f3532212e68f25da91c30c87a0d0849cf88fb63d">+2/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>badger_repository_signature_stat.go</strong><dd><code>Refactor signature stat repository to use transaction helpers</code></dd></summary>
<hr>

internal/client/repository/badger/badger_repository_signature_stat.go

• Replaced direct database transaction call with <code>DoUpdateInTx</code> method<br> • <br>Added context usage by retrieving transaction from context using <br><code>getTxn(ctx)</code>

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-1d7655d09b1154f30797eba94b35fafb24136d4e10befcd77a6c879b25db80f9">+3/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>entity.go</strong><dd><code>Add aggregation status entity and sort active validators</code>&nbsp; </dd></summary>
<hr>

core/entity/entity.go

• Modified <code>GetActiveValidators</code> to return sorted validators by operator <br>address<br> • Added new <code>AggregationStatus</code> struct with voting power and <br>validators fields

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-c76f58ee92ae2683d15cba8895a561ca38c5ea42e937e4d819ec38a73d2df3d2">+7/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>prettylog.go</strong><dd><code>Add signature listener component to logging colors</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pkg/log/prettylog.go

• Added "SIGN_LISTENER" component to green color logging alongside <br>"AGGREGATOR"

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-7b0f51ba7256131394bdae60d76c0267019f92ee8bf01cce193a58636a165e56">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>entity_error.go</strong><dd><code>Add validator not found error constant</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

core/entity/entity_error.go

• Added new error constant <code>ErrValidatorNotFound</code> for validator not <br>found scenarios

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-49035ef136e73b08d503791987de81791a0ff4e90172ca6aac4ebdc723e8bc36">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>config.go</strong><dd><code>Move config types from internal entity to local definitions</code></dd></summary>
<hr>

cmd/relay/config.go

• Moved <code>CMDCrossChainAddress</code> and <code>KeyStore</code> type definitions from <br>internal entity package to local config<br> • Updated config struct to use <br>local types instead of external entity types

</details>

  </td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-e8c6ba868ebf52e360af9fd3ec6340d1015f8644c58cfc3415f1f63394591eb6">+30/-22</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Additional files</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>entity.go</strong></td>
  <td><a href="https://github.com/symbioticfi/relay/pull/147/files#diff-f74cddfd87dccff5755030aa8e9c135db745c6284f093bd547fc7d574fc7ec94">+0/-38</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: core/entity/entity_validator_map.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/go-errors/errors"
+)
+
+type ValidatorMap struct {
+	RequestHash         common.Hash
+	Epoch               uint64
+	ActiveValidatorsMap map[common.Address]struct{}
+	IsPresent           map[common.Address]struct{}
+	QuorumThreshold     VotingPower
```
[the remark this answers is not in the document]
Comment (ilyasymbiotic) [reply core/entity/entity_validator_map.go]: Removed QuorumThreshold and TotalVotingPower, but TotalVotingPower we can only calculate from ValidatorSet currently. Will get from Header once we have this field there.
